CopyKeys.ps1

```diff
@@ -890,7 +892,7 @@ function Create-Secret(
 ### <return name="CompletedList">List of VMs for which CopyKeys ran successfully</return>
 function Start-CopyKeys
 {
-    $SuppressOutput = Login-AzureRmAccount -ErrorAction Stop
+    $SuppressOutput = connect-AzAccount -ErrorAction Stop

```

Comment:
Just a suggestion as we were planning on adding this change here.
Could you add another check here to prevent Login calls when unnecessary.
 
    $Context = Get-AzContext

    if($Context -eq $null)
    {
        $SuppressOutput = Login-AzAccount -ErrorAction Stop
    }
`

Reply:
Could you make this change as well.

Reply:
Could you please brief which change

Reply:
As per the code, the user is asked to login again and again on subsequent script runs which is not required as the context is still retained.
So, instead of the following code line, which helps login,
`$SuppressOutput = connect-AzAccount -ErrorAction Stop`
we can check for the context and run the login command only when necessary. This can be done as shown below
```
$Context = Get-AzContext

if($Context -eq $null)
{
    $SuppressOutput = Login-AzAccount -ErrorAction Stop
}
```
So, please do the needful.